trade-finance-manager-ui/component-tests/case/facility/_macros/facility_details.component-test.js

```diff
@@ -169,6 +180,78 @@ describe(component, () => {
       wrapper.expectText('[data-cy="facility-tenor"]').toRead(params.facility.dates.tenor);
     });
+
+    describe('when facility end date feature flag is not enabled', () => {
+      beforeEach(() => {
+        wrapper = render({ ...params, facility: facilityWithFacilityEndDate, isTfmFacilityEndDateFeatureFlagEnabled: false });
+      });
+      it('should not render facilityEndDateExists', () => {
+        wrapper.expectText('[data-cy="facility-end-date"]').notToExist();
+      });
+
+      it('should not render facilityEndDate', () => {
+        wrapper.expectText('[data-cy="facility-end-date"]').notToExist();
+      });
+
+      it('should not render bankReviewDate', () => {
+        wrapper.expectText('[data-cy="bank-review-date"]').notToExist();
+      });
+    });
+
+    describe('when facility end date feature flag is enabled', () => {
+      describe('when the facility has no end date information', () => {
+        beforeEach(() => {
+          wrapper = render({ ...params, isTfmFacilityEndDateFeatureFlagEnabled: true });
+        });
+        it('should render a dash for facilityEndDateExists', () => {
+          wrapper.expectText('[data-cy="facility-end-date-exists"]').toRead('-');
+        });
+
+        it('should not render facilityEndDate', () => {
+          wrapper.expectText('[data-cy="facility-end-date"]').notToExist();
+        });
+
+        it('should not render bankReviewDate', () => {
+          wrapper.expectText('[data-cy="bank-review-date"]').notToExist();
+        });
+      });
+
+      describe('when the facility has an end date', () => {
+        beforeEach(() => {
+          wrapper = render({ ...params, facility: facilityWithFacilityEndDate, isTfmFacilityEndDateFeatureFlagEnabled: true });
+        });
+        it('should render facilityEndDateExists', () => {
+          wrapper.expectText('[data-cy="facility-end-date-exists"]').toRead('Yes');
+        });
+
+        it('should render facilityEndDate', () => {
+          const expected = localiseTimestamp(facilityWithFacilityEndDate.dates.facilityEndDate, 'd MMMM yyyy', params.user.timezone);
+          wrapper.expectText('[data-cy="facility-end-date"]').toRead(expected);
+        });
+
+        it('should not render bankReviewDate', () => {
+          wrapper.expectText('[data-cy="bank-review-date"]').notToExist();
+        });
+      });
+
+      describe('when the facility does not have an end date', () => {
+        beforeEach(() => {
+          wrapper = render({ ...params, facility: facilityWithBankReviewDate, isTfmFacilityEndDateFeatureFlagEnabled: true });
+        });
+        it('should render facilityEndDateExists', () => {
+          wrapper.expectText('[data-cy="facility-end-date-exists"]').toRead('No');
+        });
+
+        it('should render facilityEndDate', () => {
+          wrapper.expectText('[data-cy="facility-end-date"]').notToExist();
+        });
+
+        it('should not render bankReviewDate', () => {
```

Comment:
Do we need bank review date tests? :)

Reply:
That was a typo there in the test name, in that it should have been 'should render bankReviewDate' - now fixed. (I think that's your question, as opposed to do we need them at all, in which case the answer is yes as they're linked to the facility end dates) :)